charts/licensify/templates/nginx-configmap.yaml

```diff
@@ -0,0 +1,134 @@
+      # https://serverfault.com/a/598106
+      map $upstream_http_strict_transport_security $strict_transport_security {
+        "" "max-age=31536000; preload";
+      }
+      map $upstream_http_permissions_policy $permissions_policy {
+        "" "interest-cohort=()";
+      }
+      map $upstream_http_x_content_type_options $x_content_type_options {
+        "" "nosniff";
+      }
+
+      log_format json_event escape=json '{'
+        '"@timestamp":"$time_iso8601",'
+        '"body_bytes_sent":$body_bytes_sent,'
+        '"bytes_sent":$bytes_sent,'
+        '"govuk_request_id":"$govuk_request_id",'
+        '"http_host":"$http_host",'
+        '"http_referer":"$http_referer",'
+        '"http_user_agent":"$http_user_agent",'
+        '"http_x_forwarded_for":"$http_x_forwarded_for",'
+        '"remote_addr":"$remote_addr",'
+        '"remote_user":"$remote_user",'
+        '"request_method":"$request_method",'
+        '"request_time":$request_time,'
+        '"request_uri":"$request_uri",'
+        '"sent_http_content_type":"$sent_http_content_type",'
+        '"sent_http_location":"$sent_http_location",'
+        '"server_protocol":"$server_protocol",'
+        '"status":$status,'
+        '"upstream_addr":"$upstream_addr",'
+        '"upstream_response_time":"$upstream_response_time"'
+      '}';
+
+      {{- if $.Values.nginx.extraHttpConf }}
+      {{ $.Values.nginx.extraHttpConf | nindent 4 }}
+      {{- end }}
+
+      upstream {{ .name }} {
+        server 127.0.0.1:{{ .port }};
+      }
+
+      server {
+        listen {{ $.Values.nginx.port }};
+        proxy_connect_timeout {{ $.Values.nginx.proxyConnectTimeout }};
+        proxy_read_timeout    {{ $.Values.nginx.proxyReadTimeout }};
+
+        access_log /dev/stdout json_event;
+        error_log /dev/stderr;
+
+        # Where the response header is already set on the incoming response,
+        # these are no-ops. https://serverfault.com/a/598106
+        add_header Strict-Transport-Security $strict_transport_security;
+        add_header Permissions-Policy $permissions_policy;
+        add_header X-Content-Type-Options $x_content_type_options always;
+
+        add_header X-Robots-Tag "noindex";
+
+        location / {
+          proxy_set_header   Host $http_host;
+          proxy_set_header   X-Real-IP $remote_addr;
+          proxy_set_header GOVUK-Request-Id $govuk_request_id;
+          proxy_pass         http://{{ .name }};
+          proxy_redirect     off;
+          client_max_body_size {{ $.Values.nginx.clientMaxBodySize }};
+        }
+
+        # Endpoint that isn't cached, which is used to assert that an external
+        # service can receive a response from GOV.UK origin on www hostname. It
+        # is intended for pingdom monitoring
+        location = /__canary__ {
+          default_type application/json;
+          add_header cache-control "max-age=0,no-store,no-cache";
+          return 200 '{"message": "Tweet tweet"}\n';
+        }
+
```

Comment:
I think we can probably drop this bit — `/__canary__` was a www.gov.uk thing and was never used for probing Licensify.